RELEASES.md

```diff
@@ -1,3 +1,113 @@
+Version 1.38.0 (2019-09-26)
+==========================
+
+Language
+--------
+- [The `#[global_allocator]` attribute can now be used in submodules.][62735]
+- [The `#[deprecated]` attribute can now be used on macros.][62042]
+
+Compiler
+--------
+- [Added support for the `aarch64-uwp-windows-msvc`, `i686-uwp-windows-gnu`,
+  `i686-uwp-windows-msvc`, `x86_64-uwp-windows-gnu`, and
+  `x86_64-uwp-windows-msvc` targets.][60260]
+- [Added support for the `armv7-unknown-linux-gnueabi` and
+  `armv7-unknown-linux-musleabi` targets.][63107]
+- [Added support for the `hexagon-unknown-linux-musl` target.][62814]
+- [Added pipelined compilation support to `rustc`.][62766] This will
+  improve compiliation times in some cases. For further information please refer
+  to the [_"Evaluating pipelined rustc compilation"_][pipeline-internals] thread.
+- [Added support for the `riscv32i-unknown-none-elf` target.][62784]
+
+
+Libraries
+---------
+- [`ascii::EscapeDefault` now implements `Clone` and `Display`.][63421]
+- [Derive macros for prelude traits (e.g. `Clone`, `Debug`, `Hash`) are now
+  available at the same path as the trait.][63056] (e.g. The `Clone` derive macro
+  is available at `std::clone::Clone`). This also makes all built-in macros
+  available in `std`/`core` root. e.g. `std::include_bytes!`.
+- [`str::Chars` now implements `Debug`.][63000]
+- [`slice::{concat, connect, join}` now accepts `&[T]` in addition to `&T`.][62528]
+- [`*const T` and `*mut T` now implement `marker::Unpin`.][62583]
+- [`Arc<[T]>` and `Rc<[T]>` now implement `FromIterator<T>`.][61953]
+- [Added euclidean remainder and division operations (`div_euclid`,
+  `rem_euclid`) to all numeric primitives.][61884] Additionally `checked`,
+  `overflowing`, and `wrapping` versions are available for all
+  integer primitives.
+- [`thread::AccessError` now implements `Clone`, `Copy`, `Eq`, `Error`, and
+  `PartialEq`.][61491]
+- [`iter::{StepBy, Peekable, Take}` now implement `DoubleEndedIterator`.][61457]
+
+Stabilized APIs
+---------------
+- [`<*const T>::cast`]
+- [`<*mut T>::cast`]
+- [`Duration::as_secs_f32`]
+- [`Duration::as_secs_f64`]
+- [`Duration::div_duration_f32`]
+- [`Duration::div_duration_f64`]
+- [`Duration::div_f32`]
+- [`Duration::div_f64`]
+- [`Duration::from_secs_f32`]
+- [`Duration::from_secs_f64`]
+- [`Duration::mul_f32`]
+- [`Duration::mul_f64`]
+- [`any::type_name`]
+
+Cargo
+-----
+- [Added pipelined compilation support to `cargo`.][cargo/7143]
+- [You can now pass the `--features` option multiple times to enable
+  multiple features.][cargo/7084]
+
+Misc
+----
+- [`rustc` will now warn if you create an invalid value, such as a null
+  reference with `mem::{uninitialized, zeroed}`.][63346]
```

Comment:
Probably some more loose language is warranted here, e.g. "for some invalid values" or something like that.

Reply:
cc @RalfJung @Centril (perhaps?) -- do we expect that these warnings will eventually migrate to errors? AFAIK, they're always UB, right?

Reply:
I think this bit should explicitly state that these cases are UB.

> AFAIK, they're always UB, right?

Yep. The lint is a conservative underestimate of known UB.

Reply:
However, the phrasing "will warn if you create an invalid value" promises too much IMO.

"Will now detect some cases of invalid values" or so is better, I think. We want to clearly communicate that this is conservative and there are many many cases of invalid values that the lint will not catch.

Reply:
I've updated the wording, let me know if it matches what you expect.

Reply:
Hm, that still sounds like we *are* catching all known cases of UB...

What about something like

> `rustc` will now warn about some incorrect uses of `mem::{uninitialized, zeroed}` that cause undefined behavior.